src/lsm/compaction.zig

```diff
@@ -584,7 +622,13 @@ pub fn CompactionType(
 
             // TODO(Beat Pacing) This should really be where the compaction callback is invoked,
             // but currently that can occur multiple times per beat.
-            compaction.grid.forfeit(compaction.grid_reservation);
+            switch (compaction.strategy) {
+                .table_move => assert(compaction.grid_reservation == null),
+                .table_merge => {
+                    compaction.grid.forfeit(compaction.grid_reservation.?);
+                    compaction.grid_reservation = null;
```

Comment:
Should we reset to null here? If so, do we need to clean up other state? I'm not sure that we should set to null.

Or otherwise, it's as if we're using the `grid_reservation` also as a way to assert idempotency?

Reply:
Viewed it as a way to assert other functions aren't being called incorrectly (similar to setting `.callback = null`)

Reply:
> Viewed it as a way to assert other functions aren't being called incorrectly (similar to setting .callback = null)

Perfect; I agree that resetting `grid_reservation` immediately after forfeit is a good practice. It guards against accidentally "leaking" a reservation between compactions.

Reply:
Yes, agreed that it's a guard. I was wondering though whether we should use it as an _implicit_ guard (as we do here), or whether we should have an _explicit_ guard for this purpose?

@sentientwaffle agreed also with your argument around accidentally "leaking" a reservation across compactions.

That settles it. Let's keep using `null`.